crates/kornia-tensor-ops/src/ops.rs

```diff
@@ -151,4 +292,118 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_dot_product1_f32() -> Result<(), TensorOpsError> {
+        let a_f32 =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[1.0, 2.0, 3.0], CpuAllocator)?;
+        let b_f32 =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[4.0, 5.0, 6.0], CpuAllocator)?;
+        let result = dot_product1(&a_f32, &b_f32)?;
+        assert_eq!(result, 32.0);
+        Ok(())
+    }
+
+    #[test]
+    fn test_dot_product1_u8() -> Result<(), TensorOpsError> {
+        let a_u8 = Tensor::<u8, 1, CpuAllocator>::from_shape_slice([3], &[1, 2, 3], CpuAllocator)?;
+        let b_u8 = Tensor::<u8, 1, CpuAllocator>::from_shape_slice([3], &[4, 5, 6], CpuAllocator)?;
+        let result = dot_product1(&a_u8, &b_u8)?;
+        assert_eq!(result, 32);
+        Ok(())
+    }
+
+    #[test]
+    fn test_cosine_similarity_orthogonal_vectors() -> Result<(), TensorOpsError> {
+        let a =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[1.0, 0.0, 0.0], CpuAllocator)?;
+        let b =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[0.0, 1.0, 0.0], CpuAllocator)?;
+        let result = cosine_similarity(&a, &b)?;
+        assert_eq!(result, 0.0);
+        Ok(())
+    }
+
+    #[test]
+    fn test_cosine_similarity_opposite_vectors() -> Result<(), TensorOpsError> {
+        let e =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[1.0, 2.0, 3.0], CpuAllocator)?;
+        let f = Tensor::<f32, 1, CpuAllocator>::from_shape_slice(
+            [3],
+            &[-1.0, -2.0, -3.0],
+            CpuAllocator,
+        )?;
+        let result = cosine_similarity(&e, &f)?;
+        assert!((result - -1.0).abs() < 1e-6);
+        Ok(())
+    }
+
+    #[test]
+    fn test_cosine_similarity_zero_vector() -> Result<(), TensorOpsError> {
+        let zero =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[0.0, 0.0, 0.0], CpuAllocator)?;
+        let g =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[1.0, 2.0, 3.0], CpuAllocator)?;
+        let result = cosine_similarity(&zero, &g)?;
+        assert_eq!(result, 0.0);
+        Ok(())
+    }
+
+    #[test]
+    fn test_cosine_similarity_zero_dot_product() -> Result<(), TensorOpsError> {
+        let h =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[1.0, 1.0, 0.0], CpuAllocator)?;
+        let i =
+            Tensor::<f32, 1, CpuAllocator>::from_shape_slice([3], &[-1.0, 1.0, 0.0], CpuAllocator)?;
+        let result = cosine_similarity(&h, &i)?;
+        assert!(result.abs() < 1e-6);
+        Ok(())
+    }
+
+    #[test]
+    fn test_cosine_similarity_3d_tensors() -> Result<(), TensorOpsError> {
+        let a = Tensor::<f32, 3, CpuAllocator>::from_shape_slice(
+            [2, 2, 2],
+            &[1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0],
```

Comment:
now seeing this test, I think we should limit also to Tensor1 this operator, as here i don't think it makes a lot of sense as it's applying to the whole tensor instead to a specific axis dimension ?